src/OMSimulatorLib/OMSimulator.cpp

```diff
@@ -154,6 +155,17 @@ oms_status_enu_t oms3_getElement(const char* cref_, oms3_element_t** element)
   return oms3::Scope::GetInstance().getElement(oms3::ComRef(cref_), reinterpret_cast<oms3::Element**>(element));
 }
 
+oms_status_enu_t oms3_addConnector(const char *cref, oms_causality_enu_t causality, oms_signal_type_enu_t type)
+{
+  logTrace();
+  oms3::ComRef tail(cref);
+  oms3::ComRef modelCref = tail.pop_front();
+  oms3::ComRef systemCref = tail.pop_front();
+  oms3::Model* model = oms3::Scope::GetInstance().getModel(modelCref);
+  oms3::System* system = model->getSystem(systemCref);
```

Comment:
Please check if `model` or `system` are NULL and give some nice error messages. E.g.

```c
if (!model)
  return logError("Model \"" + std::string(front) + "\" does not exist in the scope");
```